### DISCUSSION TITLE
add 'copy' button on developer tools/service response field

### DISCUSSION THREAD

**Discussion by Mariusthvdb** [home-assistant/frontend#18831] [Dashboard] [answered]
with the new service
```
service: weather.get_forecasts
target:
  entity_id: weather.buienradar
data:
  type: daily
response_variable: buienradar_forecast
```

we need to template the output to something like

```
        state: >
          {{buienradar_forecast['weather.buienradar'].forecast[0].temperature}}
```

to be able to test that, we need to copy the output of the service into a yaml to json converter. That process would be helped enormously by adding a 'Copy' button to the Response (Antwoord) field:

<img width="1261" alt="Scherm­afbeelding 2023-11-30 om 11 33 36" src="https://github.com/home-assistant/frontend/assets/33354141/71cd9776-d8e6-4c07-8b47-3b0b08392637">

please consider, thanks

**Comment by karwosts** [comment]
Maybe something like this would be helpful for you?

https://github.com/home-assistant/frontend/pull/18866

**Reply by ildar170975** [reply]
Unfortunately, this seems to [work in the same session](https://github.com/home-assistant/frontend/pull/18866#issuecomment-1837196442) only if I understood you correctly.
Imho a "Copy" button will be a useful addition to your prosed functionality.

**Reply by Mariusthvdb** [reply]
> Maybe something like this would be helpful for you?
> 
> 
> 
> https://github.com/home-assistant/frontend/pull/18866

Yes that would be helpful indeed!

Yet, it does not take away the request for a copy button on the actual response field.

It can be taken elsewhere for other purposes too ( eg document different services and outcomes into some comment section or the likes outside backend in one's own annotations)

Also. It seems the import button on the template tool is not persistent or very clear as to which output it is going to paste.

Suppose that would only work reliably in one sequence of operations, directly after issuing the service command?

So in short : yes, very nice indeed and thanks for that. 
But please also add the copy button, like we have elsewhere in dev tools (states)

**Comment by karwosts** [comment, marked as answer]
Implemented as requested https://github.com/home-assistant/frontend/pull/18872

**Reply by Mariusthvdb** [reply]
thank you!
